components/renderers/jsonforms-renderers.ts

```diff
@@ -72,17 +85,21 @@ module JSONForms {
         private schema: SchemaElement;
         private validationService: IValidationService;
         private pathResolver: IPathResolver;
+        private ruleService: IRuleService;
         private scope: ng.IScope;
 
-        constructor(schemaPath: string, services: JSONForms.Services, label?: string) {
+        constructor(private schemaPath: string, services: JSONForms.Services, label?: string, public rule?:IRule) {
             this.instance = services.get<JSONForms.IDataProvider>(ServiceId.DataProvider).getData();
             this.schema = services.get<JSONForms.ISchemaProvider>(ServiceId.SchemaProvider).getSchema();
             this.validationService = services.get<JSONForms.IValidationService>(ServiceId.Validation);
             this.pathResolver = services.get<JSONForms.IPathResolverService>(ServiceId.PathResolver).getResolver();
+            this.ruleService = services.get<JSONForms.IRuleService>(ServiceId.RuleService);
             this.scope = services.get<JSONForms.IScopeProvider>(ServiceId.ScopeProvider).getScope();
 
             this.path = PathUtil.normalize(schemaPath);
             this.label = this.createLabel(schemaPath, label);
+            this.ruleService.addRuleTrack(this);
+            this.ruleService.revaluateRules(this, this.schemaPath);
```

Comment:
Does this handle the init case? If so, I think we can remove this line, since a broadcast `modelChanged` event is triggered after the first `render` call.